pallets/rc-migrator/src/lib.rs

```diff
@@ -0,0 +1,2581 @@
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub mod accounts;
+pub mod claims;
+pub mod crowdloan;
+pub mod indices;
+pub mod multisig;
+pub mod preimage;
+pub mod proxy;
+pub mod referenda;
+pub mod staking;
+pub mod types;
+pub mod vesting;
+pub mod weights;
+pub mod weights_ah;
+pub use pallet::*;
+pub mod asset_rate;
+#[cfg(feature = "runtime-benchmarks")]
+pub mod benchmarking;
+pub mod bounties;
+pub mod child_bounties;
+pub mod conviction_voting;
+#[cfg(feature = "kusama-ahm")]
+pub mod recovery;
+pub mod scheduler;
+#[cfg(feature = "kusama-ahm")]
+pub mod society;
+pub mod treasury;
+pub mod xcm_config;
+
+pub use weights::*;
+
+use crate::{
+	accounts::MigratedBalances,
+	types::{MigrationFinishedData, XcmBatch, XcmBatchAndMeter},
+};
+use accounts::AccountsMigrator;
+use child_bounties::ChildBountiesMigrator;
+use claims::{ClaimsMigrator, ClaimsStage};
+use frame_support::{
+	pallet_prelude::*,
+	sp_runtime::traits::AccountIdConversion,
+	storage::transactional::with_transaction_opaque_err,
+	traits::{
+		fungible::{Inspect, InspectFreeze, Mutate, MutateFreeze, MutateHold},
+		schedule::DispatchTime,
+		tokens::{Fortitude, Pay, Precision, Preservation},
+		Contains, Defensive, DefensiveTruncateFrom, EnqueueMessage, LockableCurrency,
+		ReservableCurrency, VariantCount,
+	},
+	weights::{Weight, WeightMeter},
+	PalletId,
+};
+use frame_system::{pallet_prelude::*, AccountInfo};
+use indices::IndicesMigrator;
+use multisig::MultisigMigrator;
+use pallet_balances::AccountData;
+use pallet_message_queue::ForceSetHead;
+use polkadot_parachain_primitives::primitives::Id as ParaId;
+use polkadot_runtime_common::{
+	claims as pallet_claims, crowdloan as pallet_crowdloan, impls::VersionedLocatableAsset,
+	paras_registrar, slots as pallet_slots,
+};
+use preimage::{
+	PreimageChunkMigrator, PreimageLegacyRequestStatusMigrator, PreimageRequestStatusMigrator,
+};
+use proxy::*;
+use referenda::ReferendaStage;
+use runtime_parachains::{
+	hrmp,
+	inclusion::{AggregateMessageOrigin, UmpQueueId},
+};
+use sp_core::{crypto::Ss58Codec, H256};
+use sp_runtime::{
+	traits::{BadOrigin, BlockNumberProvider, Hash, One, Zero},
+	AccountId32, Saturating,
+};
+use sp_std::prelude::*;
+use staking::{
+	bags_list::{BagsListMigrator, BagsListStage},
+	delegated_staking::{DelegatedStakingMigrator, DelegatedStakingStage},
+	nom_pools::{NomPoolsMigrator, NomPoolsStage},
+};
+use storage::TransactionOutcome;
+use types::IntoPortable;
+pub use types::{MigrationStatus, PalletMigration, QueuePriority as AhUmpQueuePriority};
+use vesting::VestingMigrator;
+use weights_ah::WeightInfo as AhWeightInfo;
+use xcm::prelude::*;
+use xcm_builder::MintLocation;
+
+/// The log target of this pallet.
+pub const LOG_TARGET: &str = "runtime::rc-migrator";
+
+/// Soft limit on the DMP message size.
+///
+/// The hard limit should be about 64KiB (TODO: @ggwpez test) which means that we stay well below
```

Comment:
I verified some of these constraints in https://docs.google.com/document/d/1l2COWct1f-gC8nM0tq7Xs8pBWeAP6pX0LociWC6enUg/edit?tab=t.0, for any other reviewer.